website_sale_slides_order_line_link/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _action_confirm(self):
+        course_lines = self.order_line.filtered(
+            lambda line: line.product_id.channel_ids
+        ).ids
```

Comment:
You can pass the recordset as it is to simplify the logic later
```suggestion
        )
```